mbed_host_tests/host_tests_runner/mbed_base.py

```diff
@@ -70,19 +75,92 @@ def __init__(self, options):
+                    # 'DETAILS.TXT file not found
+                    else:
+                        return None
+
+                except OSError as e:
+                    self.logger.prn_err("Failed to get remount count due to OSError.", str(e))
+                    self.logger.prn_inf("Retrying in 1 second (try %s of %s)" % (cur_try, tries))
+                    sleep(1)
+            # Failed to get remount count
+            return None
+
+        def check_flash_error(target_id, disk, initial_remount_count):
+            """! Check for flash errors
+            @return Returns false if FAIL.TXT present, else true
+            """
+            if not target_id:
+                self.logger.prn_wrn("Target ID not found: Skipping flash check and retry")
+                return True
+
+            bad_files = Set(['FAIL.TXT'])
+            # Re-try at max 5 times with 0.5 sec in delay
+            for i in range(5):
+                # mbed_lstools.create() should be done inside the loop. Otherwise it will loop on same data.
+                mbeds = mbed_lstools.create()
+                mbeds_by_tid = mbeds.list_mbeds_by_targetid()   # key: target_id, value mbedls_dict()
+
+                if target_id in mbeds_by_tid:
+                    if 'mount_point' in mbeds_by_tid[target_id] and mbeds_by_tid[target_id]['mount_point']:
+                        if not initial_remount_count is None:
+                            new_remount_count = get_remount_count(disk)
+                            if not new_remount_count is None and new_remount_count == initial_remount_count:
+                                sleep(0.5)
+                                continue
+
+                        common_items = []
+                        try:
+                            items = Set([x.upper() for x in os.listdir(mbeds_by_tid[target_id]['mount_point'])])
+                            common_items = bad_files.intersection(items)
+                        except OSError as e:
+                            print "Failed to enumerate disk files, retrying"
+                            continue
+
+                        for common_item in common_items:
+                            full_path = os.path.join(mbeds_by_tid[target_id]['mount_point'], common_item)
+                            self.logger.prn_err("Found %s"% (full_path))
+                            bad_file_contents = "[failed to read bad file]"
+                            try:
+                                with open(full_path, "r") as bad_file:
+                                    bad_file_contents = bad_file.read()
+                            except IOError as error:
+                                self.logger.prn_err("Error opening '%s': %s" % (full_path, error))
+
+                            self.logger.prn_err("Error file contents:\n%s" % bad_file_contents)
+                        if common_items:
+                            return False
+                sleep(0.5)
+            return True
```

Comment:
In reference to my above comment, this function would then return the "success" flag here. This is important because it is possible that the loop will go through each of its iterations without making it to the end (for instance, by hitting the `continue` statements above).